File: src/main/java/com/moovie/backend/domain/usecase/movies/GetTrendingMovies.java
package com.moovie.backend.domain.usecase.movies;

import com.moovie.backend.domain.repository.MoviesRepository;
import com.moovie.backend.domain.model.MovieListing;
import org.springframework.stereotype.Service;

@Service
public class GetTrendingMovies {

    private final MoviesRepository moviesRepository;

    public GetTrendingMovies(MoviesRepository moviesRepository) {
        this.moviesRepository = moviesRepository;
    }

    public MovieListing invoke(int page) {
        return moviesRepository.getTrendingMovies(page);
    }
}

File: src/main/kotlin/org/mobyle/domain/usecase/movies/GetTrendingMovies.kt
package org.mobyle.domain.usecase.movies

import kotlinx.coroutines.runBlocking
import org.mobyle.domain.repository.MoviesRepository
import org.mobyle.model.MovieListing

class GetTrendingMovies(private val repository: MoviesRepository) {
    operator fun invoke(page: Int): MovieListing = runBlocking {
        repository.getTrendingMovies(page)
    }
}
